service/src/main/java/org/apache/griffin/core/job/JobServiceImpl.java

```diff
@@ -666,9 +668,31 @@ public void triggerJobById(Long id) throws SchedulerException {
                     .forJob(jobKey)
                     .startNow()
                     .build();
-            scheduler.scheduleJob(trigger);
+            TriggerKey key = trigger.getKey();
+            CountDownLatch latch = new CountDownLatch(1);
+            String listenerName = "listenerJob_" + jobKey.toString();
+            try {
+                scheduler.getListenerManager().addTriggerListener(
+                        new CountDownTriggerListener(latch, listenerName)
+                        , key::equals);
+
+                scheduler.scheduleJob(trigger);
+
+                try {
+                    latch.await(0, TimeUnit.MILLISECONDS);
+                } catch (InterruptedException e) {
+                    LOGGER.warn("CountDownLatch awaiting is interrupted");
+                }
+                List<JobInstanceBean> instanceBeans = instanceRepo.findByTriggerKey(key.toString());
```

Comment:
We probably can slightly change this API to become properly asynchronous - we can return triggerKey from the `/trigger` request and then client can send further requests with this key to find the instance as in this line.